File: tests/xcalablemp/NOT-IMPLEMENTED/291/bcast2.f90
      program main
      include 'xmp_lib.h'
      integer,parameter:: N=1000
!$xmp nodes p(*)
!$xmp template t(N,N)
!$xmp distribute t(*,cyclic) onto p
      integer aa(N), a
      real*4 bb(N), b
      real*8 cc(N), c
      integer procs, result

      procs = xmp_num_nodes()

      do j=1, procs
         a = xmp_node_num()
         b = real(a)
         c = dble(a)
         do i=1, N
            aa(i) = a+i-1
            bb(i) = real(a+i-1)
            cc(i) = dble(a+i-1)
         enddo

!$xmp bcast (a) from p(j) on t(:,:)
!$xmp bcast (b) from p(j) on t(:,:)
!$xmp bcast (c) from p(j) on t(:,:)
!$xmp bcast (aa) from p(j) on t(:,:)
!$xmp bcast (bb) from p(j) on t(:,:)
!$xmp bcast (cc) from p(j) on t(:,:)

         result = 0
         if(a .ne. j) result = -1
         if(b .ne. real(j)) result = -1
         if(c .ne. dble(j)) result = -1
         do i=1, N
            if(aa(i) .ne. j+i-1) result = -1
            if(bb(i) .ne. real(j+i-1)) result = -1
            if(cc(i) .ne. dble(j+i-1)) result = -1
         enddo
      enddo

!$xmp reduction(+:result)
!$xmp task on p(1)
  if( result .eq. 0 ) then
     write(*,*) "PASS"
  else
     write(*,*) "ERROR"
     call exit(1)
  endif
!$xmp end task

      end

File: tests/xcalablemp/global-view/bcast/C/bcast2.c
#include <xmp.h>
#include <stdlib.h>
#include <stdio.h>
#pragma xmp nodes p(4,*)
static const int N=1000;
int i,j,k,aa[1000],a;
double bb[1000],b;
float cc[1000],c;
int procs,procs2,ans,result=0;

int main(void)
{
  procs = xmp_num_nodes();
  procs2 = procs/4;

  a = xmp_node_num();
  b = (double)a;
  c = (float)a;
  for(i=0;i<N;i++){
    aa[i] = a+i;
    bb[i] = (double)(a+i);
    cc[i] = (float)(a+i);
  }
#pragma xmp bcast (a)
#pragma xmp bcast (b)
#pragma xmp bcast (c)
#pragma xmp bcast (aa)
#pragma xmp bcast (bb) 
#pragma xmp bcast (cc) 
  ans = 1;
  if(a != ans) result = -1;
  if(b != (double)ans) result = -1;
  if(c != (float)ans) result = -1;
  for(i=0;i<N;i++){
    if(aa[i] != ans+i) result = -1;
    if(bb[i] != (double)(ans+i)) result = -1;
    if(cc[i] != (float)(ans+i)) result = -1;
  }

#pragma xmp reduction(+:result)
#pragma xmp task on p(1,1)
  {
    if(result == 0){
      printf("PASS\n");
    }
    else{
      fprintf(stderr, "ERROR\n");
      exit(1);
    }
  }

  return 0;
}    
